healthcareai/supervised_model_trainer.py

```diff
@@ -47,6 +47,26 @@ def __init__(self, dataframe, predicted_column, model_type, impute=True, grain_c
         # Split the data into train and test
         self._advanced_trainer.train_test_split()
 
+        # Save the original dataframe columns to the parent class
+        self._advanced_trainer.pre_dummified_columns = dataframe.columns.values
+
+        # Identify the categorical columns
```

Comment:
Re 1: My understanding was that all of the data cleaning -- dummification in particular -- happens before the advanced trainer is ever used.  Is there a good way to move this functionality to the advanced trainer without also pushing all the data cleaning into the advanced trainer?